lib/services/redfish-validator-service.js

```diff
@@ -207,6 +211,34 @@ function redfishValidatorFactory(
         return Promise.reject(new Errors.NotFoundError ('Message registry not found'));
 
     };
+    RedfishValidator.prototype.getVendorNameById = function(identifier) {
+        var result = {node: undefined, vendor: undefined};
+        return waterline.nodes.getNodeById(identifier)
+        .then(function(node){
+	    if(!node){
+	        throw new Errors.NotFoundError('invalid node id.');
+	    }
+	    result.node = node;
+	    result.vendor = _getSystemVendorByIdentifiers(node.identifiers);
```

Comment:
You don't have to declare a variable in advance, and also use `needOne` rather than `getNodeById` will save a throw statement. For simplicity, you could just do as following:
```javascript
RedfishValidator.prototype.getVendorNameById = function(nodeId) {
    return waterline.nodes.needOne({id: nodeId})
        .then(function(node){
	        return {
                node: node;
	            vendor: _getSystemVendorByIdentifiers(node.identifiers);
            }
	});
};
```